File: xRayLum.cpp
#include <iostream>
#include <stdio.h>
#include <list>
#include <fstream>
#include <math.h>
#include <string>
#include <vector>

using namespace std;

vector<double> xRayLum(double ucInput, double uc, double rad, double shockRad)
{
	//Open the data file
	ifstream file ("data");
	if(!file.is_open())
	{
		cout << "Unable to open data file";
	}
	
	//Take the data in the data file and place it into respective lists
	//Units:
	//		radius: cm
	//		rho: g/cm^3
	//		temp: K
	vector<double> radius, rho, temp, velocity, soundSpeed, junk3;
	double num;
	int i = 0;
	while (file >> num) {
		if(i == 0)
		{
			radius.push_back(num);
			i = 1;
		}
		else if(i == 1)
		{
			velocity.push_back(num);
			i = 2;
		}
		else if(i == 2)
		{
			soundSpeed.push_back(num);
			i = 3;
		}
		else if(i == 3)
		{
			rho.push_back(num);
			i = 4;
		}
		else if(i == 4)
		{
			junk3.push_back(num);
			i = 5;
		}
		else if(i == 5)
		{
			temp.push_back(num);
			i = 0;
		}
	}
	file.close();
	
	//Open the analyticCooling.txt file and get the xSpec data
	ifstream file2 ("cool05_8_Teague.txt");
	if(!file2.is_open())
	{
		cout << "Unable to open analyticCooling.txt file";
	}
	
	//Units:
	//		xSpecT: K
	//		xSpecLambda: (erg/s)*cm^3
	vector<double> xSpecT, xSpecLambda, junk4;
	double num1;
	int index = 0;
	
	//Get rid of the first line of the file
	//string firstline;
	//getline(file2, firstline);
	
	while (file2 >> num1) {
		if(index == 0)
		{
			xSpecT.push_back(pow(10., num1));
			index = 1;
		}
		else if(index == 1)
		{
			junk4.push_back(num1);
			index = 2;
		}
		else if(index == 2)
		{
			xSpecLambda.push_back(pow(10., num1));
			index = 0;
		}
	}
	file2.close();
	
	//Constants
	double MASS = 1.021*pow(10,-24); //grams
	
	//Units: numParticles/cm^3
	vector<double> numDensity;
	for(vector<double>::size_type i=0; i<rho.size(); i++)
	{
		numDensity.push_back(rho[i]/MASS);
	}

	//Calculate the cooling function for each value of T in temp.
	//Must use the data given in xSpecT and xSpecLambda to estimate the
	//correct values for lambda at given T
	vector<double> coolLambda;
	
	//Loop through every term in the temp list
	double tempVal, leftT, rightT, leftLambda, rightLambda, estLambda;
	for(vector<double>::size_type i=0; i<temp.size(); i++)
	{
		tempVal = temp[i];
		//For each term in temp estimate a value for lambda by 
		//finding which two temperatures the temp term falls between in the xSpecT list.
		//Then find the average between the two lambdas that the two xSpecT temperatures
		//correspond to.
		for(vector<double>::size_type s=0; s<xSpecT.size()-1; s++)
		{
			leftT = xSpecT[s];
			rightT = xSpecT[s+1];
			if(tempVal >= leftT && tempVal <= rightT)
			{
				leftLambda = xSpecLambda[s];
				rightLambda = xSpecLambda[s+1];
				estLambda = (leftLambda + rightLambda)/2;
				break;
			}
		}
		coolLambda.push_back(estLambda);
	}
	
	//Numerically integrate the luminosity function:
	// 	Lx = integral((1.2)(0.71^2)*(n^2)*(r^2)*(lambda)dr) from 0 to infinity
	//Also integrate the luminosity function but without the r^2 term to give the emissivity
	vector<double> integrandAll;
	vector<double> emissivity;
	for(vector<double>::size_type i=0; i<radius.size(); i++)
	{
		integrandAll.push_back((1.2)*(pow(0.71, 2))*(pow(numDensity[i],2))*(pow(radius[i],2))*(coolLambda[i]));
		emissivity.push_back((1.2)*(pow(0.71, 2))*(pow(numDensity[i],2))*(coolLambda[i]));
	}
	
	//Calculate area under curve by adding up consecutive rectangles
	int i2 = 0, stepSize = 1;
	double totalArea = 0;
	double x1, x2, y2, tempArea, y1, yAvg;
	int vecSize = integrandAll.size();
	//double pc2cm = 3.086*pow(10,18);
	//double newDistLim = 753.487 * pc2cm;
	while(true)
	{
		x1 = radius[i2];
		x2 = radius[i2+stepSize];
		y2 = integrandAll[i2+stepSize];
		y1 = integrandAll[i2];
		yAvg = (y2+y1)/2;
		tempArea = (x2-x1)*(yAvg);
		totalArea += tempArea;
		i2 += stepSize;
		if((i2+stepSize) >= vecSize){break;}
		//if(radius[i2+stepSize] >= newDistLim){break;}
	}
	cout<<"X-ray luminosity within 200,000pc : "<<totalArea<<" erg/s\n";
	
	vector<double> returnVec;
	returnVec.push_back(totalArea);
	
	//Create a vector that contains the value of the integral for different radii
	vector<double> integralSolved;
	double lumAtRadius;
	
	for(vector<double>::size_type i=0; i<radius.size(); i++)
	{
		lumAtRadius = 0;
		for(vector<double>::size_type s=0; s<i; s++)
		{
			lumAtRadius += (radius[s+1]-radius[s])*(0.5*((integrandAll[s+1])+(integrandAll[s])));
			if(isinf(lumAtRadius))
			{
				lumAtRadius = 0;
			}
		}
		integralSolved.push_back(lumAtRadius);
	}
	
	////////////////////////////////////////////////////////////////////////////////////
	//Perform integration with a stepsize = 1pc instead of 1 index in the radius vector.
	////////////////////////////////////////////////////////////////////////////////////
	/*
	vector<double> numDen1pc;
	vector<double> cool1pc;
	vector<double> rad1pc;
	
	//Must have a value for radius from 5-200,000pc
	double pc2cm = 3.086*pow(10,18);
	for(int i=6; i<200000; i++)
	{
		rad1pc.push_back(i*pc2cm);
	}
	
	//Must have a value for number density and lambda at each radii from 5-200,000pc
	double currRad;
	for(vector<double>::size_type i=0; i<rad1pc.size(); i++)
	{
		currRad = rad1pc[i];
		
		int normIndex;
		int splitIndex;
		for(vector<double>::size_type s=0; s<radius.size()-1; s++)
		{
			double leftRad = radius[s];
			double rightRad = radius[s+1];
			if(currRad > leftRad && currRad < rightRad)
			{
				normIndex = s;
				
				double diffStepSize = (rightRad - leftRad)/10;
				for(int g = 0; g < 10; g++) {
					if (currRad > (leftRad + g*diffStepSize) && currRad < (leftRad + (g+1)*diffStepSize)) {
						splitIndex = g;
					}
				}
				
			}
		}
		
		//Get new number density
		
		double stepSizeLambda = (coolLambda[normIndex+1] - coolLambda[normIndex])/10;
		double newLambda = coolLambda[normIndex] + splitIndex*stepSizeLambda;
		cool1pc.push_back(newLambda);
		
		//Get new cooling Lambda
		
		double stepSizeNumDen = (numDensity[normIndex+1] - numDensity[normIndex])/10;
		double newNumDen = numDensity[normIndex] + splitIndex*stepSizeNumDen;
		numDen1pc.push_back(newNumDen);
	}
	
	//Numerically integrate the luminosity function:
	// 	Lx = integral((1.2)(0.71^2)*(n^2)*(r^2)*(lambda)dr) from 0 to infinity
	//Also integrate the luminosity function but without the r^2 term to give the emissivity
	vector<double> integrandAll1pc;
	vector<double> emissivity1pc;
	for(vector<double>::size_type i=0; i<rad1pc.size(); i++)
	{
		integrandAll1pc.push_back((1.2)*(pow(0.71, 2))*(pow(numDen1pc[i],2))*(pow(rad1pc[i],2))*(cool1pc[i]));
		emissivity1pc.push_back((1.2)*(pow(0.71, 2))*(pow(numDen1pc[i],2))*(cool1pc[i]));
	}
	
	//Calculate area under curve by adding up consecutive rectangles
	int i3 = 0, stepSize1 = 1;
	double totalArea1pc = 0;
	double x11, x22, y22, tempArea1, y11, yAvg1;
	int vecSize1 = integrandAll1pc.size();
	while(true)
	{
		x11 = rad1pc[i3];
		x22 = rad1pc[i3+stepSize1];
		y22 = integrandAll1pc[i3+stepSize1];
		y11 = integrandAll1pc[i3];
		yAvg1 = (y22+y11)/2;
		tempArea1 = (x22-x11)*(yAvg1);
		totalArea1pc += tempArea1;
		i3 += stepSize1;
		if((i3+stepSize1) >= vecSize1){break;}
	}
	
	//Create a vector that contains the value of the integral for different radii
	vector<double> integralSolved1pc;
	double lumAtRadius1;
	
	for(vector<double>::size_type i=0; i<rad1pc.size(); i++)
	{
		lumAtRadius1 = 0;
		for(vector<double>::size_type s=0; s<i; s++)
		{
			lumAtRadius1 += (rad1pc[s+1]-rad1pc[s])*(0.5*((integrandAll1pc[s+1])+(integrandAll1pc[s])));
			if(isinf(lumAtRadius1))
			{
				lumAtRadius1 = 0;
			}
		}
		integralSolved1pc.push_back(lumAtRadius1);
	}
	
	//Open a file to store each run's data
	string tempStr_1pc = to_string(ucInput);
	tempStr_1pc = tempStr_1pc.substr(0, 5);
	string fileName2 = "DataFolder/testStepData_ucInput" + tempStr_1pc + ".txt";
	ofstream b_file (fileName2);
	b_file << scientific;
	if(b_file.is_open())
	{
		//Set up categories for file
 		b_file << "   " << "Radius" << "\t\t " << "NumDensity" << "\t\t " << "CoolLambda" << "\t\t ";
		b_file << "X-ray Lum" << "\t\t " << "Emissivity" << "\t\t " << "Integrand";
		b_file << "\t\t\t" << "uc = " << uc;
		b_file << endl;
	}
	else cout << "Unable to open file";
	
	for(vector<double>::size_type i=0; i<rad1pc.size(); i++)
	{
		b_file << rad1pc[i] << "    " << numDen1pc[i] << "    " << cool1pc[i] << "    ";
		b_file << integralSolved1pc[i] << "    " << emissivity1pc[i] << "    ";
		b_file << integrandAll1pc[i];
		b_file << endl;
	}
	b_file.close();
	*/
	////////////////////////////////////////////////////////////////////////////////////
	////////////////////////////////////////////////////////////////////////////////////
	
	//Open a file to store each run's data
	string tempStr = to_string(ucInput);
	tempStr = tempStr.substr(0, 5);
	string fileName = "DataFolder/ucInput" + tempStr + ".txt";
	ofstream a_file (fileName);
	a_file << scientific;
	if(a_file.is_open())
	{
		//Set up categories for file
 		a_file << "   " << "Radius" << "\t\t    " << "rho" << "\t\t\t" << "Temperature" << "\t\t  ";
		a_file << "X-ray Lum" << "\t\t " << "Emissivity" << "\t\t     " << "u" << "\t\t         " << "cs";
		a_file << "\t\t\t" << "uc = " << uc;
		a_file << endl;
	}
	else cout << "Unable to open file";
	
	//To add the integrad to the file like before: add integrandAll[i]
	for(vector<double>::size_type i=0; i<radius.size(); i++)
	{
		a_file << radius[i] << "    " << rho[i] << "    " << temp[i] << "    ";
		a_file << integralSolved[i] << "    " << emissivity[i] << "    ";
		a_file << velocity[i] << "    " << soundSpeed[i] << "    " << integrandAll[i];
		a_file << endl;
	}
	a_file.close();
	
	//Find and print the maximum velocity
	double maxVel = -2; 
	double tempVel = -1;
	double radiusInParsec;
	double radiusLimit = 2.5*rad;
	for(vector<double>::size_type i=0; i<radius.size(); i++)
	{
		radiusInParsec = radius[i]/(3.086*pow(10,18));
		if(radiusInParsec > radiusLimit){break;}
		else
		{
			tempVel = velocity[i];
			if(tempVel > maxVel)
			{
				maxVel = tempVel;
			}
		}
	}
	maxVel = maxVel/100000;
	cout << "Max Velocity within " << to_string(radiusLimit).substr(0,6) << "pc: " << maxVel << " km/s\n";
	
	//Find and print the asymptotic velocity
	double asymVel = -2; 
	double tempVel2 = -1;
	double radiusInParsec2;
	double radiusLimit2 = 150000;
	for(vector<double>::size_type i=0; i<radius.size(); i++)
	{
		radiusInParsec2 = radius[i]/(3.086*pow(10,18));
		if(radiusInParsec2 > radiusLimit2){break;}
		else
		{
			tempVel2 = velocity[i];
			if(tempVel2 > asymVel)
			{
				asymVel = tempVel2;
			}
		}
	}
	asymVel = asymVel/100000;
	cout << "Aysmptotic Velocity: " << asymVel << " km/s\n";
	
	//Print the central temperature
	cout << "Central Temperature: " << temp[0] << " K\n";
	
	fileName = "DataFolder/Extra_Data" + tempStr + ".txt";
	ofstream test_file (fileName);
	test_file << scientific;
	for(vector<double>::size_type i=0; i<radius.size(); i++)
	{
		test_file << radius[i] << "    " << rho[i] << "    " << temp[i] << "    ";
		test_file << integralSolved[i] << "    " << emissivity[i] << "    ";
		test_file << velocity[i] << "    " << soundSpeed[i] << "    " << integrandAll[i] << "    ";
		test_file << numDensity[i] << "    " << coolLambda[i];
		test_file << endl;
	}
	test_file.close();
	
	return returnVec;
	
	//Add code to find the radius with 50% of the luminosity
}

File: xRayLum.py
import matplotlib.pyplot as plt
import sys
import math
import numpy
from statistics import mode

#Open the data file
file = open('data','r')
radius = []
rho = []
temp = []

#Take the data in the data file and place it into respective lists
#Units:
#		radius: cm
#		rho: particles/cm^3
#		temp: K
while True:
    s = file.readline()
    if not s: break
    radius.append(float(s[0:13]))#/(3.086*10**18))
    rho.append(float(s[39:52]))
    temp.append(float(s[65:78]))
file.close()

#Open the analyticCooling.txt file and get the xSpec data
file = open('analyticCooling.txt','r')
xSpecT =[]
xSpecLambda = []
#Units:
#		xSpecT: K
#		xSpecLambda: erg/s cm^3
xf=file.readline()
while True:
	xf=file.readline()
	if not xf: break
	xSpecT.append(10**(float(xf[2:7])))
	xSpecLambda.append(10**(float(xf[21:35])))
file.close()

#Constants
MASS = 1.021*10**-24 #grams

#Units: numParticles/cm^3
numDensity = [x/MASS for x in rho]
#nSubH = [0.71*x for x in numDensity]
#nSube = [1.2*x for x in nSubH]

#Calculate the cooling function for each value of T in temp.
#Must use the data given in xSpecT and xSpecLambda to estimate the 
#correct values for lambda at given T.
coolLambda = []

#Loop through every term in the temp list
for i in range(len(temp)):
	tempVal = temp[i]
	#For each term in temp estimate a value for Lambda by 
	#finding which two temperatures the temp term falls between in the xSpecT list.
	#Then find the average between the two lambdas that the two xSpecT temperatures 
	#correspond to.
	for s in range(len(xSpecT)-1):
		leftT = xSpecT[s]
		rightT = xSpecT[s+1]
		if(tempVal >= leftT and tempVal <= rightT):
			leftLambda = xSpecLambda[s]
			rightLambda = xSpecLambda[s+1]
			estLambda = (leftLambda + rightLambda)/2
			break
	coolLambda.append(estLambda)
#plt.plot(temp, coolLambda, "b")
#plt.show()

stepsizes, xraylums = [], []
#Create a while loop to allow for integration over many different ranges
#Step size corresponds to number of points skipped in radius list. 
#The average difference between points in the radius list is 21pc
stepsize = 1
while True:
	#Get user input
	if(True):
		while True:
			try:
				lowerBoundpc = float(input("Enter lower bound in parsecs: "))
				if(lowerBoundpc >= 0): break
				else: print("Must input an integer >= 0")
			except ValueError: print("Must input an integer")
		while True:
			try:
				upperBoundpc = float(input("Enter upper bound in parsecs: "))
				if(upperBoundpc >= lowerBoundpc): break
				else: print("Must input an integer > the lower bound")
			except ValueError: print("Must input an integer")
	
	#lowerBoundpc = 0
	#upperBoundpc = 3500
	
	#Convert upper and lower bounds into cm
	lowerBound = lowerBoundpc * (3.086*10**18)
	upperBound = upperBoundpc * (3.086*10**18)
	#Numerically integrate the luminosity function:
	#	Lx = integral((1.2)(0.71^2)*(n^2)*(r^2)*(lambda)dr) from 0 to infinity
	integrandAll, integrandCustom = [], []
	xAxisAll, xAxis = [], []
	newRadius, newRadiusAll = [], []
	#stepsize = 5
	for i in range(0, len(radius), stepsize):
		integrandAll.append((1.2)*(0.71**2)*(numDensity[i]**2)*(radius[i]**2)*(coolLambda[i]))
		xAxisAll.append(radius[i]/(3.086*10**18))
		newRadiusAll.append(radius[i])
		if(radius[i] >= lowerBound and radius[i] <= upperBound):
			integrandCustom.append((1.2)*(0.71**2)*(numDensity[i]**2)*(radius[i]**2)*(coolLambda[i]))
			xAxis.append(radius[i]/(3.086*10**18))
			newRadius.append(radius[i])
###############################################################################################################
	#Calculate area under curve by adding up consecutive rectangles		
	i, totalArea = 0, 0
	while True:
		x1 = newRadiusAll[i]
		x2 = newRadiusAll[i+1]
		y2 = integrandAll[i+1]
		tempArea = (x2-x1)*(y2)
		totalArea = tempArea + totalArea
		i = i+1
		if((i+1) >= len(integrandAll)): break
	print('X-ray luminosity within 200,000pc: {:.3e}' .format(totalArea),'erg/s')
	
	#Plot the integrand over the entire range for the radius
	if(True):
		plt.plot(xAxisAll, integrandAll, "b")
		plt.xlabel("Radius (pc)")
		plt.ylabel("X-ray Luminosity (erg/(cm*s)")
		plt.title("X-ray Emission of a Galactic Wind")
		ax = plt.subplot(111)
		plt.text(0.95, 0.95, 'X-ray luminosity within 200,000pc:\n{:.3e}erg/s' .format(totalArea),
			fontsize = 14, color = 'red', transform=ax.transAxes,
			horizontalalignment='right', verticalalignment='top')
		plt.show()
	
###############################################################################################################
	
	#Calculate the area under the curve by adding up consecutive rectangles
	i, area = 0, 0
	while True:
		x1 = newRadius[i]
		x2 = newRadius[i+1]
		y2 = integrandCustom[i+1]
		tempArea = (x2-x1)*(y2)
		area = tempArea + area
		i = i+1
		if((i+1)>=len(integrandCustom)): break
	percentArea = (area/totalArea)*100
	print('X-ray luminosity within',upperBoundpc,'pc: {:.3e}'.format(area), "erg/s")
	print("Percent of total within: %.2f%%" % (percentArea))
	
	#Plot the integrand over the range given by the user
	if(stepsize == 60 or stepsize == 80 or stepsize == 1):
		plt.plot(xAxis, integrandCustom, "b")
		plt.xlabel("Radius (pc)")
		plt.ylabel("X-ray Luminosity (erg/(cm*s)")
		plt.title("Integrand")
		ax = plt.subplot(111)
		plt.text(0.95, 0.95, 'X-ray luminosity within %s pc:\n{:.3e}erg/s'
			'\nPercent of total luminosity:\n%.2f%%\nStep Size: %.1f' 
			.format(area) % (upperBoundpc, percentArea, stepsize),
			fontsize = 14, color = 'red', transform=ax.transAxes,
			horizontalalignment='right', verticalalignment='top')
		plt.show()
	
###############################################################################################################

	#Create a vector that contains the value of the integral for different radii
	integralSolved = []

	for p in range(0, len(newRadius)):
		lumAtRadius = 0
		w = 0
		while(w < p):
			lumAtRadius = lumAtRadius + (newRadius[w+1]-newRadius[w])*(integrandCustom[w+1])
			if(numpy.isinf(lumAtRadius)):
				lumAtRadius = 0
			w = w + 1
		integralSolved.append(lumAtRadius)
	
	#Plot the luminosity over the range given by the user
	if(stepsize == 60 or stepsize == 80 or stepsize == 1):
		plt.xscale('log')
		plt.plot(xAxis, integralSolved, "b")
		plt.xlabel("Radius (pc)")
		plt.ylabel("X-ray Luminosity (erg/s)")
		plt.title("X-ray Luminosity")
		ax = plt.subplot(111)
		plt.text(0.95, 0.95, 'Step Size: %.1f' % (stepsize),
			fontsize = 14, color = 'red', transform=ax.transAxes,
			horizontalalignment='right', verticalalignment='top')
		plt.show()
	
###############################################################################################################
	
	stepsizes.append(stepsize)
	xraylums.append(totalArea)
	
	if(stepsize >= 100): break
	if(stepsize == 1): stepsize = 2
	else: stepsize = stepsize + 1
	
	# keepGoing = input('Enter "exit" to stop program; Press enter to begin again: ').lower()
	# if(keepGoing == "exit"): break

#Write the stepsize data to a file	
outf = open("stepSizeData.txt", "w")
for i in range(len(stepsizes)):
	outf.write("{:03d}  " .format(stepsizes[i]))
	outf.write("{}\n" .format(xraylums[i]))
outf.close()
plt.plot(stepsizes, xraylums, "b")
plt.xlabel("Step Size")
plt.ylabel("Total X-ray Luminosity")
plt.title("Effect of Step Size")
#plt.show()
plt.close()

#Find the mean, median, mode of the differences between consecutive elements in the radius list
# rDiffs = []
# for j in range(len(radius) - 1):
	# tempVal1 = radius[j]
	# tempVal2 = radius[j+1]
	# rDiffs.append(abs(tempVal1 - tempVal2))	
# print("Mean: ", numpy.mean(rDiffs)/(3.086*10**18))
# print("Median: ", numpy.median(rDiffs)/(3.086*10**18))
# print("Mode: ", mode(rDiffs)/(3.086*10**18))

###############################################################################################################

#Calculate Luminosity with increments of 1pc using averages
#Range of radius list goes from 5pc - 199980pc

# #Write the two lists to a file to use in MatLab
# matLab = open("integrandPoints.txt", "w")
# for i in range(len(newRadiusAll)):
	# matLab.write("{},  " .format(newRadiusAll[i]))
	# matLab.write("{}\n" .format(integrandAll[i]))
# matLab.close()

myRadii = []
radPC = 5
while True:
	radCM = radPC * (3.086*10**18)
	myRadii.append(radCM)
	radPC = radPC + 1
	if(radPC >= 3501): break
	
fullIntegrand = []
for i in range(len(radius)):
	fullIntegrand.append((1.2)*(0.71**2)*(numDensity[i]**2)*(radius[i]**2)*(coolLambda[i]))

myIntegrand = []
incrementSize = 100
for j in range(len(myRadii)):
	curRad = myRadii[j]
	for k in range(len(radius)-1):
		if(curRad >= radius[k] and curRad <= radius[k+1]):
			ystepValues, xstepValues = [], []
			deltaY = abs(fullIntegrand[k] - fullIntegrand[k+1])
			yincrement = deltaY/incrementSize
			deltaX = abs(radius[k] - radius[k+1])
			xincrement = deltaX/incrementSize
			for l in range(incrementSize+1):
				ystepValues.append(fullIntegrand[k] + l*yincrement)
				xstepValues.append(radius[k] + l*xincrement)
			for m in range(incrementSize):
				if(curRad >= xstepValues[m] and curRad <= xstepValues[m+1]):
					averageVal = abs(ystepValues[m] + ystepValues[m+1])/2
					myIntegrand.append(averageVal)
					break

myRadiiPC = []
for r in range(len(myRadii)):
	myRadiiPC.append(myRadii[r]/(3.086*10**18))
plt.plot(myRadiiPC, myIntegrand, "b")
plt.xlabel("Radius (pc)")
plt.ylabel("Integrand")
plt.title("Integrand with Steps = 1pc")
#plt.show()
plt.close()

#Now integrate up to each radius and plot to see how luminosity changes over radius
myLum = []

for p in range(0, len(myRadii)):
	myCurrLum = 0
	n = 0
	while(n < p):
		myCurrLum = myCurrLum + (myRadii[n+1]-myRadii[n])*(myIntegrand[n+1])
		if(numpy.isinf(myCurrLum)):
			myCurrLum = 0
		n = n + 1
	myLum.append(myCurrLum)

plt.plot(myRadii, myLum, "b")
plt.xlabel("Radius (pc)")
plt.ylabel("X-ray Luminosity (erg/s)")
plt.title("X-ray Luminosity with steps = 1pc")
#plt.show()
plt.close()





























